14-draft.json

```diff
@@ -975,6 +975,69 @@
               "value"
             ]
           }
+        },
+        "network_traffic": {
+          "description": "The current network traffic, in bits/second or packets/second (or both)",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "properties": {
+                "type": "object",
+                "properties": {
+                  "bits_per_second": {
+                    "description": "",
+                    "type": "object",
+                    "properties": {
+                      "value": {
+                        "description": "The measurement value, in bits/second",
+                        "type": "number",
+                        "minimum": 0
+                      },
+                      "maximum": {
+                        "description": "The maximum available throughput in bits/second, e.g. as sold by your ISP",
+                        "type": "number",
+                        "minimum": 0
+                      }
+                    },
+                    "required": [
+                      "value"
+                    ]
+                  },
+                  "packets_per_second": {
+                    "description": "",
+                    "type": "object",
+                    "properties": {
+                      "value": {
+                        "description": "The measurement value, in packets/second",
+                        "type": "number",
+                        "minimum": 0
+                      }
+                    },
+                    "required": [
+                      "value"
+                    ]
+                  }
+                }
+              },
+              "name": {
+                "description": "Name of the measurement, e.g. to distinguish between upstream and downstream traffic",
+                "type": "string"
+              },
```

Comment:
I find it a bit strange that we combine both bits per second and packets per second measurements into the same object, but then create separate objects for upstream and downstream.

Might it make sense to either create separate measurement subtypes for bits per second and packets per second, or alternatively integrate upstream/downstream into the schema in a more structured way than a "name" field?

Note that I'm just brainstorming. None of these proposed solutions seem ideal to me...

Reply:
I did have similar ideas, though I chose this design because it felt the most consistent with the other sensor types. E.g. you'd describe the amount of your different Mate flavors (yeah, I know, Mate flavors != upstream vs downstream) like this:

```json
"beverage_supply": [
  {
    "name": "Club Mate",
    "unit": "btl",
    "value": 23,
    "location": "Fridge"
  },
  {
    "name": "Flora Mate",
    "unit": "btl",
    "value": 23,
    "location": "Fridge"
  }
]
```

In addition, it feels kinda weird to impose an artifical restriction (forcing definition of a hierarchy implied by "upstream" and "downstream) which doesn't necessarily match all real world scenarios. E.g. consider a hackerspace with a single firewall between WAN, LAN and a DMZ: While the WAN can clearly be defined as "upstream" with regard to LAN and DMZ, how do you define the relationship between LAN and DMZ? IMO there is no clear hierarchical concept defining one of the two networks as "upstream" in regard to the other.

Reply:
Yep, I think you're right.